roles/zuul/files/jobs/dib.yaml

```diff
@@ -0,0 +1,23 @@
+- job:
+    name: 'dib-build'
+    node: 'ubuntu-xenial'
+    builders:
+      shell: |
+        virtualenv $HOME/dib-build
+        . $HOME/dib-build/bin/activate
+        pip install diskimage-builder
+
+        git clone https://github.com/BonnyCI/hoist.git /opt/hoist
+        export ELEMENTS_PATH=/opt/hoist/roles/nodepool/files/etc/nodepool/elements/nodepool
+        # NOTE: be sure the list of nodepool elements here stays consistent
+        # with the elements specified in the nodepool role.
+        disk-image-create -x -t qcow2 --checksum --no-tmpfs                 \
+                          --qemu-img-options 'compat=0.10'                  \
+                          -o /opt/nodepool/images/ubuntu-xenial             \
+                          ubuntu-minimal vm simple-init growroot            \
+                          openssh-server devuser haveged pip-and-virtualenv \
+                          nodepool bonnyci-nodepool
```

Comment:
I'd love to see if we can extract the list of elements we want to test from hoist itself, if its possible.  This static list is fine for testing what we currently ship, but if we add a new element to hoist, we have no way of testing it pre-merge until the job here is updated.